pkg/commands/user.go

```diff
@@ -47,43 +46,15 @@ func (r *UserCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.Bu
 		}
 	}
 
-	// Lookup by username
-	userObj, err := user.Lookup(userStr)
+	_, _, err = util.GetUserFromUsername(userStr, groupStr)
 	if err != nil {
-		if _, ok := err.(user.UnknownUserError); ok {
-			// Lookup by id
-			userObj, err = user.LookupId(userStr)
-			if err != nil {
-				return err
-			}
-		} else {
-			return err
-		}
+		return err
 	}
 
-	// Same dance with groups
-	var group *user.Group
 	if groupStr != "" {
-		group, err = user.LookupGroup(groupStr)
-		if err != nil {
-			if _, ok := err.(user.UnknownGroupError); ok {
-				group, err = user.LookupGroupId(groupStr)
-				if err != nil {
-					return err
-				}
-			} else {
-				return err
-			}
-		}
-	}
-
-	uid := userObj.Uid
-	if group != nil {
-		uid = uid + ":" + group.Gid
+		userStr = userStr + ":" + groupStr
 	}
-
-	logrus.Infof("Setting user to %s", uid)
-	config.User = uid
+	config.User = userStr
```

Comment:
Initially this was set to uid:gid.
Now its being set to userStr:groupStr.
Is this intended ?

Reply:
Yes, that's how docker does it and we want to mimic what docker does as closely as possible.